pkg/controllers/jivavolume_controller.go

```diff
@@ -199,6 +212,136 @@ func (r *JivaVolumeReconciler) isScaleup(cr *openebsiov1alpha1.JivaVolume) bool
 	return false
 }
 
+func shouldMoveReplicas(cr *openebsiov1alpha1.JivaVolume) bool {
+	if cr.Spec.Policy.Target.ReplicationFactor < 3 {
+		return false
+	}
+	availableReplicas := 0
+	qurom := (cr.Spec.Policy.Target.ReplicationFactor / 2) + 1
+	for _, rep := range cr.Status.ReplicaStatuses {
+		if rep.Mode == "RW" {
+			availableReplicas += 1
+			if availableReplicas == qurom {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (r *JivaVolumeReconciler) moveReplicasForMissingNodes(cr *openebsiov1alpha1.JivaVolume) error {
+
+	// if the volume does not HA replicas in
+	// RW mode skip the process
+	if !shouldMoveReplicas(cr) {
+		return nil
+	}
+
+	var (
+		replicaLabel   = "openebs.io/component=jiva-replica,openebs.io/persistent-volume="
+		nodeAnnotation = "volume.kubernetes.io/selected-node"
+	)
+	labelSelector, err := labels.Parse(
+		replicaLabel + cr.Name)
+	if err != nil {
+		return err
+	}
+	pods := corev1.PodList{}
+	err = r.List(context.TODO(), &pods, &client.ListOptions{
+		Namespace:     cr.Namespace,
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		pvc := &corev1.PersistentVolumeClaim{}
+		err = r.Get(context.TODO(),
+			types.NamespacedName{
+				Name:      pod.Spec.Volumes[0].PersistentVolumeClaim.ClaimName,
+				Namespace: pod.Namespace,
+			}, pvc)
+		if err != nil {
+			// if the PVC is missing then only
+			// delete the sts pod
+			if errors.IsNotFound(err) {
+				err = r.Delete(context.TODO(), &pod)
+				// wait for pod to get deleted and
+				// recreated
+				time.Sleep(5 * time.Second)
+				if err != nil {
+					return err
+				}
```

Comment:
Do we need to handle not found case?
```suggestion
				if err != nil && !errors.IsNotFound(err) {
					return err
				}
```